pkg/kubelet/kubelet_node_status.go

```diff
@@ -579,13 +580,17 @@ func (kl *Kubelet) tryUpdateNodeStatus(ctx context.Context, tryNumber int) error
 	}
 
 	node, changed := kl.updateNode(ctx, originalNode)
-	shouldPatchNodeStatus := changed || kl.clock.Since(kl.lastStatusReportTime) >= kl.nodeStatusReportFrequency
+	shouldPatchNodeStatus := changed || kl.clock.Since(kl.lastStatusReportTime) >= kl.nextNodeStatusReportInterval
 
 	if !shouldPatchNodeStatus {
 		kl.markVolumesFromNode(node)
 		return nil
 	}
 
+	// The nextNodeStatusReportInterval will range from 75% to 125% of the specified nodeStatusReportFrequency.
+	kl.nextNodeStatusReportInterval = time.Duration(float64(kl.nodeStatusReportFrequency) * (0.75 + 0.5*rand.Float64()))
```

Comment:
if you initialize the value here the first time it will be zero, no? so `shouldPatchNodeStatus` will always trigger

Reply:
`kl.nextNodeStatusReportInterval` is first initialized in `pkg/kubelet/kubelet.go`. See:
https://github.com/kubernetes/kubernetes/pull/128394/files#diff-67dd9e8f3ee257072765326cb4f242852554a2c0753563fa51e292c0a63a7b94R562